packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -123,7 +123,7 @@ const createFSMachine = (
   return interpret(fsMachine).start()
 }
 
-exports.sourceNodes = (api, pluginOptions) => {
+exports.sourceNodesStatefully = (api, pluginOptions) => {
```

Comment:
this is breaking change - this bumps minimal supported gatsby version, so it probably should mean major version bump for `gatsby-source-filesystem`

Reply:
Maybe we should move `gatsby-source-filesystem` changes to a separate follow-up PR?

Reply:
That's good idea - I'll add TO-DO to description. Just don't want to split it right now to not maintain 2 branches (until drupal case (see my latest comments) is figured out)